packages/labs/task/src/task.ts

```diff
@@ -285,9 +285,14 @@ export class Task<
         return renderer.complete?.(this.value!);
       case TaskStatus.ERROR:
         return renderer.error?.(this.error);
-      default:
-        // exhaustiveness check
-        this.status as void;
+      default: {
+        // exhaustiveness check that can get optimized out
+        if (false as boolean) {
+          const never: never = this.status;
+          console.error(`Unexpected task status: ${never}`);
```

Comment:
Why the `console.error()`? This can never be reached?

Reply:
Good point, rewrote to still fail if the type isn't `never` at this point, but in a lighter weight way